k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp

```diff
@@ -219,21 +223,43 @@ StatusCode MarlinProcessorWrapper::initialize() {
 }
 
 StatusCode MarlinProcessorWrapper::execute() {
+
+  // Get Event
   info() << "Getting the event for " << m_processor->name() << endmsg;
   DataObject* pObject = nullptr;
   StatusCode  sc      = eventSvc()->retrieveObject("/Event/LCEvent", pObject);
+
+  lcio::LCEventImpl* the_event = nullptr;
+
   if (sc.isFailure()) {
-    error() << "Failed to retrieve the LCEvent " << endmsg;
-    return sc;
+    the_event = new lcio::LCEventImpl();
+    // Register empty event
+    debug() << "Registering conversion EDM4hep to LCIO event in TES" << endmsg;
+    auto pO = std::make_unique<LCEventWrapper>(the_event);
```

Comment:
I think we still have to add the "ownership" or "cleanup" parameter to the LCEventWrapper to indicate of the_event should be cleaned in the end or not. Depending on if the event came from the LCReader or is created like this here.